### PR TITLE
build with OpenMP by default in setup.py

### PR DESCRIPTION
**Pull request by akshayka**
This change removes the OpenMP compilation flag from the setup.py installer. The installer now attempts to build with OpenMP -- if the compiler being used does not implement the specification, then the installation will proceed without OpenMP. The installer still does not handle Windows, however.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -75,6 +77,59 @@ def run_install():
...
+    # clean up
+    os.chdir(curdir)
+    shutil.rmtree(tmpdir)
+    return exit_code == 0
+
+
+class build_ext_scs(build_ext):
+    def build_extensions(self):
+        # fetch the name of the compiler to be invoked
+        cc = self.compiler.compiler_so[0]
```
Comment (bodono) [inline python/setup.py]: It appears that this line is breaking the build on windows:

https://ci.appveyor.com/project/bodono/scs/build/1.0.304/job/3uqr6dxjy759wkyt

"AttributeError: MSVCCompiler instance has no attribute 'compiler_so'"

**Comment by bodono** [comment]
Is there any reason to not link -lgomp? It's in there because I ran into issues without it on some linux boxes I tested on.

**Comment by akshayka** [comment]
Thanks for comments -- I'll add lgomp back in and look into the Windows build, and I'll leave another comment once things are working.

**Comment by akshayka** [comment]
Looking at the output for the failing tests, it seems like the DLLs for _scs_direct and _scs_indrect are built. I suspect that the _"DLL load failed: The specified module could not be found."_ error is symptomatic of missing OpenMP DLLs on the test environment. In particular, [OpenMP DLLs](https://msdn.microsoft.com/en-us/library/0h7x01y0.aspx?tduid=(195a430e843030ac295cb6c9708522dc)(256380)(2459594)(XdSn0e3h3.k-7pDs_0iLSoAD3vZ45haRKw)()) are in the Visual C++ redistributable directory, which probably doesn't come packaged with the Microsoft Visual C++ Compiler for Python. So, even though the compiler supports the '/openmp' flag, my guess is that the DLLs themselves aren't available to it.

Is it possible to verify whether the test environment has the Visual C++ redistributable directory?

**Comment by bodono** [comment]
Strangely it passes the python3 tests, for both 64 and 32 bit...

Perhaps the best option would be to try and detect if the openmp dlls are installed before trying to run the setup. I see that you compile a test program to check for openmp, can you also try and run the program in the compilation script? If compiling or running the test program fails, then we can fall back to no openmp support with a warning message about it.

**Comment by akshayka** [comment]
Okay, that sounds like a good, quick fix. I'll implement it later this week and ping you once it's tested.

**Comment by akshayka** [comment]
The python 2.7 tests were still failing on Windows, even with the test program compilation and execution; the test program would successfully compile and run, but the DLL load error was still encountered when scs_direct/scs_indirect were imported. I spoke with Steven Diamond and he suggested that we ignore OpenMP on Windows for now, which is what I've implemented in this latest update.

**Comment by bodono** [comment]
Thanks for this! (And sorry for the delay in merging).

**Pull request merged**